department-of-veterans-affairs/next-build
# Social Links Refactor

## Description
Relates to #[16379](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16379). (or closes?)

Refactors social links to account for facilities, events, and VBA social links. 

Changes:
- Removes old SocialLinksEvents component and test
- Refactors SocialLinks component to account for all current variations
- Updates tests to account for different variations
- Updates layouts to use new components
- Adds SocialItem component to account for possible future added link types
- Unrelated:
  - Updated a unit test in events.test.ts that was failing for me due to date formatting


## Testing done
Local

## Screenshots
<img width="816" alt="Screen Shot 2024-01-18 at 10 53 37 AM" src="https://github.com/department-of-veterans-affairs/next-build/assets/61624970/ec0528d8-5eab-4c84-849e-cd73aef379d9">
<img width="816" alt="Screen Shot 2024-01-18 at 10 53 44 AM" src="https://github.com/department-of-veterans-affairs/next-build/assets/61624970/d1bfb05e-6b8b-4060-bf99-851595cbd31d">
<img width="816" alt="Screen Shot 2024-01-18 at 10 54 11 AM" src="https://github.com/department-of-veterans-affairs/next-build/assets/61624970/5fd2b707-3e25-4992-af4b-c28965b9c3b1">


## QA steps
- Navigate to a story page and verify links render as expected, ie:  http://localhost:3000/butler-health-care/stories/celebrating-our-family-caregivers/
- Navigate to an events page and verify links render as expected, ie: http://localhost:3000/central-iowa-health-care/events/52265/
- VBA is not currently available in next-build, but you could test by just modifying the events component to include vba props

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:
